lib/najax.js

```diff
@@ -168,28 +159,35 @@ function najax (uri, options, callback) {
       jqXHR.status = statusCode
 
       if (isSuccess) {
-        // Set data for the fake xhr object
-        jqXHR.statusText = statusText
         // success, statusText, jqXHR
-        dfd.resolve(data, statusText, jqXHR)
+        dfd.resolve(data, jqXHR.statusText, jqXHR)
       } else {
         // jqXHR, statusText, error
         // When an HTTP error occurs, errorThrown receives the textual portion of the
         // HTTP status, such as "Not Found" or "Internal Server Error."
-        statusText = 'error'
-        dfd.reject(jqXHR, statusText, http.STATUS_CODES[statusCode])
+        jqXHR.statusText = 'error'
+        onError(new Error(http.STATUS_CODES[statusCode]))
```

Comment:
It should be fine to pass an Error instead of a string here.  The Error generally will transform itself back into a String for any string uses, so I think we'll retain backwards compatibility.